src/org/benf/cfr/reader/bytecode/analysis/parse/expression/CastExpression.java

```diff
@@ -57,6 +63,34 @@ public Expression deepClone(CloneHelper cloneHelper) {
         return new CastExpression(getLoc(), getInferredJavaType(), cloneHelper.replaceOrClone(child), forced);
     }
 
+    @Override
+    public Literal getComputedLiteral(Map<LValue, Literal> display) {
+        if (getChild() instanceof Literal) {
+            Literal childLiteral = (Literal) getChild();
+            CastAction action = childLiteral.getValue().getInferredJavaType().chain(getInferredJavaType());
+            if (action == CastAction.None) {
+                return childLiteral;
+            }
+            Number value = (Number) childLiteral.getValue().getValue();
+            switch (getInferredJavaType().getRawType()) {
+                case BOOLEAN:
+                case CHAR:
+                case SHORT:
+                case INT:
+                case BYTE:
+                    return new Literal(TypedLiteral.getInt(value.intValue()));
```

Comment:
This will have the wrong type inside it - perhaps add a getInt(intVal, RawJavaType) to TypedLiteral